src/stan/agrad/autodiff.hpp

```diff
@@ -359,6 +358,76 @@ namespace stan {
       stan::agrad::recover_memory_nested();
     }
 
-  }
-}
+    /**
+     * Calculate the value, the Hessian, and the gradient of the Hessian
+     * of the specified function at the specified argument.
+     *
+     * <p>The functor must implement
+     *
+     * <code>
+     * using stan::agrad::fvar;
+     * using stan::agrad::var;
+     *
+     * fvar<fvar<var> >
+     * operator()(const
+     * Eigen::Matrix<fvar<fvar<var> >, Eigen::Dynamic, 1>&)
+     * </code>
+     *
+     * using only operations that are defined for
+     * <code>stan::agrad::fvar</code> and <code>stan::agrad::var</code>.
+     *
+     * This latter constraint usually
+     * requires the functions to be defined in terms of the libraries
+     * defined in Stan or in terms of functions with appropriately
+     * general namespace imports that eventually depend on functions
+     * defined in Stan.
+     *
+     * @tparam F Type of function
+     * @param[in] f Function
+     * @param[in] x Argument to function
+     * @param[out] fx Function applied to argument
+     * @param[out] H Hessian of function at argument
+     * @param[out] grad_H Gradient of the Hessian of function at argument
+     */
+    template <typename F>
+    void
+    grad_hessian(const F& f,
+                 const Eigen::Matrix<double, Eigen::Dynamic, 1>& x,
+                 double& fx,
+                 Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>& H,
+                 std::vector<Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> >& grad_H) {
+      using Eigen::Matrix;
+      using Eigen::Dynamic;
+      start_nested();
+      fx = f(x);
+      try {
+        int d = x.size();
+        H.resize(d, d);
```

Comment:
FYI for the future, we can do the vector resize and the Matrix construction at the same time with:

grad_H.resize(d, Eigen::Matrix<double,-1,-1>(d,d));